backend/src/schema/resolvers/posts.spec.js

```diff
@@ -91,44 +91,63 @@ afterEach(async () => {
 })
 
 describe('Post', () => {
-  const postQueryFilteredByCategories = gql`
-    query Post($filter: _PostFilter) {
-      Post(filter: $filter) {
-        id
-        categories {
-          id
-        }
-      }
-    }
-  `
-
-  const postQueryFilteredByEmotions = gql`
-    query Post($filter: _PostFilter) {
-      Post(filter: $filter) {
-        id
-        emotions {
-          emotion
-        }
-      }
-    }
-  `
-
   describe('can be filtered', () => {
-    let post31, post32
+    let followedUser, happyPost, cryPost
     beforeEach(async () => {
-      ;[post31, post32] = await Promise.all([
-        factory.create('Post', { id: 'p31', categoryIds: ['cat4'] }),
-        factory.create('Post', { id: 'p32', categoryIds: ['cat15'] }),
-        factory.create('Post', { id: 'p33', categoryIds: ['cat9'] }),
+      ;[followedUser] = await Promise.all([
+        factory.create('User', {
+          id: 'followed-by-me',
+          email: 'followed@example.org',
+          name: 'Followed User',
+          password: '1234',
+        }),
+      ])
+      ;[happyPost, cryPost] = await Promise.all([
+        factory.create('Post', { id: 'happy-post', categoryIds: ['cat4'] }),
+        factory.create('Post', { id: 'cry-post', categoryIds: ['cat15'] }),
+        factory.create('Post', {
+          id: 'post-by-followed-user',
+          categoryIds: ['cat9'],
+          author: followedUser,
+        }),
       ])
     })
 
+    describe('no filter', () => {
+      it('returns all posts', async () => {
+        const postQueryNoFilters = gql`
+          query Post($filter: _PostFilter) {
+            Post(filter: $filter) {
+              id
+            }
+          }
+        `
+        const expected = [{ id: 'happy-post' }, { id: 'cry-post' }, { id: 'post-by-followed-user' }]
+        variables = { filter: {} }
+        await expect(query({ query: postQueryNoFilters, variables })).resolves.toMatchObject({
+          data: {
+            Post: expect.arrayContaining(expected),
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-09-11T09:28:38Z" title="Wednesday, September 11th 2019, 11:28:38 am +02:00">Sep 11, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -91,44 +91,63 @@ afterEach(async () => {
 })
 
 describe('Post', () => {
-  const postQueryFilteredByCategories = gql`
-    query Post($filter: _PostFilter) {
-      Post(filter: $filter) {
-        id
-        categories {
-          id
-        }
-      }
-    }
-  `
-
-  const postQueryFilteredByEmotions = gql`
-    query Post($filter: _PostFilter) {
-      Post(filter: $filter) {
-        id
-        emotions {
-          emotion
-        }
-      }
-    }
-  `
-
   describe('can be filtered', () => {
-    let post31, post32
+    let followedUser, happyPost, cryPost
     beforeEach(async () => {
-      ;[post31, post32] = await Promise.all([
-        factory.create('Post', { id: 'p31', categoryIds: ['cat4'] }),
-        factory.create('Post', { id: 'p32', categoryIds: ['cat15'] }),
-        factory.create('Post', { id: 'p33', categoryIds: ['cat9'] }),
+      ;[followedUser] = await Promise.all([
+        factory.create('User', {
+          id: 'followed-by-me',
+          email: 'followed@example.org',
+          name: 'Followed User',
+          password: '1234',
+        }),
+      ])
+      ;[happyPost, cryPost] = await Promise.all([
+        factory.create('Post', { id: 'happy-post', categoryIds: ['cat4'] }),
+        factory.create('Post', { id: 'cry-post', categoryIds: ['cat15'] }),
+        factory.create('Post', {
+          id: 'post-by-followed-user',
+          categoryIds: ['cat9'],
+          author: followedUser,
+        }),
       ])
     })
 
+    describe('no filter', () => {
+      it('returns all posts', async () => {
+        const postQueryNoFilters = gql`
+          query Post($filter: _PostFilter) {
+            Post(filter: $filter) {
+              id
+            }
+          }
+        `
+        const expected = [{ id: 'happy-post' }, { id: 'cry-post' }, { id: 'post-by-followed-user' }]
+        variables = { filter: {} }
+        await expect(query({ query: postQueryNoFilters, variables })).resolves.toMatchObject({
+          data: {
+            Post: expect.arrayContaining(expected),
```


If you use `toMatchObject` you don't need the `expect.arrayContaining` unless you expect even more items than the three.

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-09-11T10:32:43Z" title="Wednesday, September 11th 2019, 12:32:43 pm +02:00">Sep 11, 2019</time>_

---


Actually, we started off by not using `arrayContaining` and the test failed because it returned the posts in a different order than our `expected` array, so we refactored it to say the order is not important.

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-09-11T11:28:35Z" title="Wednesday, September 11th 2019, 1:28:35 pm +02:00">Sep 11, 2019</time>_

---


You could `orderBy` in the query? It does not really matter. I like your approach, too! :+1: